File: examples/java/kora-java-scheduling-jdk/src/main/java/ru/tinkoff/kora/example/scheduling/jdk/OnceScheduler.java
package ru.tinkoff.kora.example.scheduling.jdk;

import java.time.temporal.ChronoUnit;
import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleOnce;

@Component
public final class OnceScheduler {

    private int state = 0;

    @ScheduleOnce(delay = 50, unit = ChronoUnit.MILLIS)
    void schedule() {
        state++;
    }

    public int getState() {
        return state;
    }
}

File: examples/kotlin/kora-kotlin-scheduling-jdk/src/main/kotlin/ru/tinkoff/kora/kotlin/example/scheduling/jdk/OnceScheduler.kt
package ru.tinkoff.kora.kotlin.example.scheduling.jdk

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleAtFixedRate
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleOnce
import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleWithFixedDelay
import java.time.temporal.ChronoUnit

@Component
class OnceScheduler {
    var state = 0
        private set

    @ScheduleOnce(delay = 50, unit = ChronoUnit.MILLIS)
    fun schedule() {
        state++
    }
}

